### ISSUE TITLE
L003: dbt macro with jinja false positive

### ISSUE THREAD

**Issue by rapidsiege** [sqlfluff/sqlfluff#3273]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Rule L003 doesn't seem to properly read indentations from dbt macros with jinja tags.

### Expected Behaviour

Would expect L003 not to trigger and pass the lint without issues. It looks like a false positive to me.

### Observed Behaviour

Take the following example:

```
with cte1 as (
    select *,
    from {{ source('source_db', 'source_table') }}
),

cte2 as (
    {{ macro('cte1', ['argument1', 'argument2']) }}
)

select *, from cte2 
```

The following lint error will show up:
`L:   7 | P:   5 | L003 | Expected 1 indentations, found 0 [compared to line 07]`

This happens across all models 

### How to reproduce

Run `sqlfluff lint` on any dbt model that contains a macro within jinja tags.

### Dialect

BigQuery

### Version

Version `0.13.0`. This model successfully passes `sqlfluff lint` on version `0.11.2` or older.

### Configuration

`.sqlfluff`
```
[sqlfluff]
dialect = bigquery
exclude_rules = L016,L026,L031,L032,L034,L054
ignore_templated_areas = True
templater = dbt
```

`.sqlfluffignore`
```
target/
# dbt <1.0.0
dbt_modules/
# dbt >=1.0.0
dbt_packages/
macros/
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by barrywhart** [comment]
This issue may be a duplicate of #3125, fixed in PR #3278.

Can you please provide the definition for the macro used in your example:
```
macro('cte1', ['argument1', 'argument2'])
```

Without this, I can't be sure I'm seeing the same issue you are.

**Comment by rapidsiege** [comment]
Hey @barrywhart thanks for the reply. The macro with issues is defined as follows:
```
{% macro macro_name(source_table, primary_key, sort_field) %}
```
Where both `primary_key` and `sort_field` arguments can either be a single parameter or an [array], which I don't know if it can potentially cause any other issue. We use this snipet of code to process them:
```
{% if primary_key is string %}
    {% set primary_key = [primary_key] %}
{% endif %}
```
And then the different elements are just called with a simple loop inside the actual macro:
```
{% for i in primary_key %}{{ i }},{% endfor %}
```
I'm not sure I fully answered your question (I hope so). If you need any more details, I will just copy the whole macro next time.


On a side note, I've also tested a couple of different things, and a macro as simple as the following also has the same L003 rule issue:

```
{% macro latest_rows(source_table, sort_field) %}

with selected_rows as (
    select *,
    from {{ source_table }}
    where sort_field = max(sort_field)
)

select *, from selected_rows
{% endmacro %}
```
With the model being something as simple as:

```
with data as (
    select *,
    from {{ source('source_database', 'example_table') }}
),

latest_data as (
    {{ latest_rows('data ','upload_timestamp') }}
)

select *, from latest_data 
```

**Comment by rapidsiege** [comment]
Issue persists in `sqlfluff` & `sqlfluff-templater-dbt` version `0.13.1`.

**Comment by rapidsiege** [comment]
Upon testing, I can confirm that changes from this commit https://github.com/sqlfluff/sqlfluff/commit/06c6182cd4ed04bc21cfa1fa6c472fd9e8373a30 fixed the issue, so I'm closing since it was a duplicate issue in the end. Thanks for everything.
